ibis/common/pretty.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import annotations
+
+import functools
+from typing import IO, Any
+
+import sqlalchemy as sa
+import sqlalchemy.dialects.mysql
+import sqlalchemy.dialects.postgresql
+from multipledispatch import Dispatcher
+from sqlalchemy.engine import Dialect
+from sqlalchemy.sql import ClauseElement
+
+import ibis
+import ibis.common.exceptions as com
+import ibis.expr.types as ir
+
+_get_compiled_string = Dispatcher("_get_compiled_string")
```

Comment:
Could we move this implementation to the base backends?

Reply:
It's kind of specific to pretty printing SQL. How about leaving it here until it becomes clear that it would be more broadly useful?